src/main/java/io/jenkins/plugins/coverage/adapter/DListingReportAdapter.java

```diff
@@ -0,0 +1,195 @@
+package io.jenkins.plugins.coverage.adapter;
+
+import com.google.common.collect.Lists;
+import hudson.Extension;
+import io.jenkins.plugins.coverage.adapter.parser.CoverageParser;
+import io.jenkins.plugins.coverage.exception.CoverageException;
+import io.jenkins.plugins.coverage.targets.CoverageElement;
+import io.jenkins.plugins.coverage.targets.CoverageResult;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import javax.annotation.Nonnull;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.util.List;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Collectors;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class DListingReportAdapter extends CoverageReportAdapter {
+
+    private static final String UNCOVERED = "0000000";
+    private static final String NON_EXECUTABLE_CODE = "";
+
+    @DataBoundConstructor
+    public DListingReportAdapter(String path) {
+
+        super(path == null || path.equals("") ? "*.lst" : path);
```

Comment:
I think it is better to not set a default path.

Reply:
The idea was to have same behavior like in https://github.com/jenkinsci/warnings-ng-plugin/blob/master/SUPPORTED-FORMATS.md
You can set a default pattern. In case of D coverage, it will always be *.lst at the project root folder.

I will change it to ```super(path);```